core/server/src/main/java/alluxio/worker/block/BlockLockManager.java

```diff
@@ -49,18 +46,6 @@
   /** The unique id of each lock. */
   private static final AtomicLong LOCK_ID_GEN = new AtomicLong(0);
 
-  /** A pool of read write locks. */
-  private final ResourcePool<ClientRWLock> mLockPool = new ResourcePool<ClientRWLock>(
-      WorkerContext.getConf().getInt(Constants.WORKER_TIERED_STORE_BLOCK_LOCKS)) {
-    @Override
-    public void close() {}
-
-    @Override
-    protected ClientRWLock createNewResource() {
-      return new ClientRWLock();
-    }
-  };
```

Comment:
@aaudiber  This is a per block lock. I think we allocate other data structures per block as well. For example, LockResource. 

Note that if we have this resource pool, it is another source of deadlock when the number of blocks sent from the heartbeat is greater than the #locks.